plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseTableProperties.java

```diff
@@ -36,26 +42,120 @@
+                        new ArrayType(VARCHAR),
+                        List.class,
+                        ImmutableList.of(),
+                        false,
+                        value -> ImmutableList.copyOf(((Collection<?>) value)),
+                        value -> value),
+                new PropertyMetadata<>(
+                        PRIMARY_KEY_PROPERTY,
+                        "columns to be the primary key. Optional",
+                        new ArrayType(VARCHAR),
+                        List.class,
+                        ImmutableList.of(),
+                        false,
+                        value -> ImmutableList.copyOf(((Collection<?>) value).stream()
+                                .map(name -> ((String) name).toLowerCase(ENGLISH))
+                                .collect(Collectors.toList())),
+                        value -> value),
+                stringProperty(
+                        SAMPLE_BY_PROPERTY,
+                        "An expression for sampling. Optional.",
+                        null,
                         false));
     }
 
-    public static Optional<String> getEngine(Map<String, Object> tableProperties)
+    public static ClickHouseEngineType getEngine(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        return (ClickHouseEngineType) tableProperties.get(ENGINE_PROPERTY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getOrderBy(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        List<String> orderBy = (List<String>) tableProperties.get(ORDER_BY_PROPERTY);
+        return formatProperty(orderBy);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getPartitionBy(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
+        List<String> partitionBy = (List<String>) tableProperties.get(PARTITION_BY_PROPERTY);
+        return formatProperty(partitionBy);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<String> getPrimaryKey(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+        List<String> primaryKey = (List<String>) tableProperties.get(PRIMARY_KEY_PROPERTY);
+        return formatProperty(primaryKey);
+    }
 
-        return Optional.ofNullable(tableProperties.get(ENGINE_PROPERTY)).map(String.class::cast);
+    public static Optional<String> getSampleBy(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+
+        return Optional.ofNullable(tableProperties.get(SAMPLE_BY_PROPERTY)).map(String.class::cast);
+    }
+
+    /**
+     * format property to match ClickHouse create table statement
+     *
+     * @param prop property will be formatted
+     * @return formatted property
+     */
+    private static Optional<String> formatProperty(List<String> prop)
```

Comment:
Is this tested exhaustively?

Reply:
do you means `formatProperty` function or table engine?

Reply:
I meant `formatProperty`. It'd be nice to have tests which try to excerise this method with possible inputs and verify that the resulting SQL is accepted by ClickHouse and that the result is what we expect.

Reply:
So, you means I need add more tests for testing `formatProperty` method?

Reply:
Yes. All the 3 possibilities. No properties, 1 property, more than 1 property and also some cases where the property includes whitespace (if ClickHouse has any such properties).

Reply:
Thanks, I will add these tests tomorrow.